go/chat/storage/storage_test.go

```diff
@@ -690,6 +705,133 @@ func TestStorageExpunge(t *testing.T) {
 	expunge(msgI.GetMessageID()+12, true)
 }
 
+func TestStoragePurgeEphemeral(t *testing.T) {
+	// Uses this conversation:
+	// A start                            <not deletable>
+	// B text                             <not ephemeral>
+	// C text <----\        edited by E
+	// D headline  |                      <not deletable>
+	// E edit -----^        edits C
+	// F text <---\         deleted by G
+	// G delete --^    ___  deletes F     <not deletable>
+	// H text           |
+
+	_, storage, uid := setupStorageTest(t, "delh")
+
+	convID := makeConvID()
+	msgA := makeMsgWithType(1, chat1.MessageType_TLFNAME)
+	msgB := makeText(2, "some text")
+	msgC := makeEphemeralText(3, "some text", &chat1.MsgEphemeralMetadata{})
+	msgD := makeHeadlineMessage(4)
+	msgE := makeEphemeralEdit(5, msgC.GetMessageID(), &chat1.MsgEphemeralMetadata{})
```

Comment:
We should enforce that an edit that is not ephemeral cannot be applied to a message that is...